### PR TITLE
Fix broken link of test integration framework

### PR DESCRIPTION
**Pull request by hi-rustin**
#### What this PR does
Fix broken link of test integration framework. The old one was already 404.
#### Which issue(s) this PR fixes or relates to
None

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]
Thank you.

File: docs/internal/contributing/how-integration-tests-work.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Mimir integration tests are written in Go and based on a [custom framework](https://github.com/grafana/mimir/tree/main/integration/e2e) running Mimir and its dependencies in Docker containers and using the Go [`testing`](https://golang.org/pkg/testing/) package for assertions. Integration tests run in CI for every PR, and can be easily executed locally during development (it just requires Docker).
+Mimir integration tests are written in Go and based on a [custom framework](https://github.com/grafana/mimir/tree/main/integration/e2emimir) running Mimir and its dependencies in Docker containers and using the Go [`testing`](https://golang.org/pkg/testing/) package for assertions. Integration tests run in CI for every PR, and can be easily executed locally during development (it just requires Docker).
```
Comment (pracucci) [inline docs/internal/contributing/how-integration-tests-work.md:1]: Ehm... the framework is actually at https://github.com/grafana/e2e. @hi-rustin do you mind opening another PR to fix it, please? :)

Comment (pstibrany) [reply docs/internal/contributing/how-integration-tests-work.md:1]: 🤦 At least the link works now :)
